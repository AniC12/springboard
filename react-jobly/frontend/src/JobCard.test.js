import '@testing-library/jest-dom';
import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import JobCard from "./JobCard";
import UserContext from "./UserContext";
import { UserProvider } from "./testUtils";


const mockApplyToJob = jest.fn();
const mockHasAppliedToJob = jest.fn();

it("matches snapshot", function () {
    let item = { title: "CEO", salary: 1000000, equity: 10 };
    const { asFragment } = render(
        <UserProvider>
            <JobCard item={item} />
        </UserProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("renders job information", () => {
    render(
        <UserContext.Provider value={{ hasAppliedToJob: mockHasAppliedToJob, applyToJob: mockApplyToJob }}>
            <JobCard id={1} title="Test Job" salary={120000} equity={0.1} companyName="Test Company" />
        </UserContext.Provider>
    );

    expect(screen.getByText("Test Job")).toBeInTheDocument();
    expect(screen.getByText("Test Company")).toBeInTheDocument();
    expect(screen.getByText(/Salary:/)).toBeInTheDocument();
    expect(screen.getByText(/Equity:/)).toBeInTheDocument();
});

it("handles apply button click", () => {
    mockHasAppliedToJob.mockReturnValue(false);
    render(
        <UserContext.Provider value={{ hasAppliedToJob: mockHasAppliedToJob, applyToJob: mockApplyToJob }}>
            <JobCard id={1} title="Test Job" salary={120000} equity={0.1} companyName="Test Company" />
        </UserContext.Provider>
    );

    fireEvent.click(screen.getByText("Apply"));
    expect(mockApplyToJob).toHaveBeenCalledWith(1);
});