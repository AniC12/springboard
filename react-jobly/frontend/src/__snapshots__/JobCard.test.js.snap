// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jobCard"
  >
     
    <div>
      <h6 />
      <p />
      <button>
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
